### PR TITLE
#208/Create a database table to persist the blocks

### LINKED ISSUE #208: Create a database table to persist the blocks
Users can block other users.

### PR DESCRIPTION
**Pull request by gbudau**
close #208

### CODE REVIEW

File: db/setup/schema.sql
Comment (gbudau) [inline db/setup/schema.sql:3]: Sorry for these formatting changes. I've installed the [Prettier SQL](https://marketplace.visualstudio.com/items?itemName=inferrinizzard.prettier-sql-vscode) extension to format SQL files, which uses the [sql-formatter](https://github.com/sql-formatter-org/sql-formatter) library.

File: db/setup/schema.sql
Comment (Damrod) [inline db/setup/schema.sql:30]: what does on delete cascade do?

> ON DELETE CASCADE clause in MySQL is used to automatically remove the matching records from the child table when we delete the rows from the parent table. It is a kind of referential action related to the foreign key.

Which is here the child table? Can you make an example, please?

**Review by Damrod — APPROVED** [review]
LGTM!

File: db/setup/schema.sql
Comment (fizcamposancos) [reply db/setup/schema.sql:30]: If I am not wrong, the child table is this one, the one with `ON DELETE CASCADE`. So, for example, if you delete a user, his auths, blocks, ... (games, friends,... when we have everything ready) would be deleted automatically "on cascade".

**Review by fizcamposancos — APPROVED** [review]
LGTM

**Comment by gbudau** [comment]
I merged all stacked PRs into this one, and will merge this into main after we merge #245

File: db/setup/schema.sql
Comment (vdedios) [reply db/setup/schema.sql:3]: I think is a good idea, I'll install it to!

File: transcendence-app/src/shared/relationship/relationship.module.ts
Comment (vdedios) [inline transcendence-app/src/shared/relationship/relationship.module.ts:10]: As stated in https://github.com/42AGV/ft_transcendence/pull/248#discussion_r1004798700 I disagree with having yet another module for this. This has no domain logic in our app, it's just a status between users.

I suspect you had this approach due to this being in a new sql table, but I don't see a problem consuming this block repository from users module. The new table is clearly tied to users :crying_cat_face:

If we fear user controller/service etc starts growing we could split controller in some files to split logic.

Comment (Damrod) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: I also don't understand where is this thing coming from. Was it here yesterday when I reviewed it?

Comment (gbudau) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: Ok, I will move this logic into the Users module.

@Damrod, this was the `BlockModule` yesterday. I renamed it to the `RelationshipModule` today to add the friendship relation in the same module.

Comment (gbudau) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: Done :rocket:
I've added this change in #258 because it's the last PR from the stack of PRs related to this feature

Comment (Damrod) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: You may think it's fun, but I've spent one hour tracking down bugs, since I decided to merge your branch into my latest branch, to try and not have merge conflicts.

Comment (gbudau) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: Sorry, @Damrod.

I think squashing the PRs does not help because if we stack PRs and a squash creates a different commit with all the commits into one, the PRs on top lose the reference to the previous commits.
What do we think about changing our workflow to a merge commit even if the git history will become less "clean"?

cc @fizcamposancos @vdedios

Comment (Damrod) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: I agree. No more squashing, as long as we have so many dependencies.

Comment (vdedios) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: I think, rather than a matter of squashing being the problem, I see the problem is indeed working over other's ongoing work when it has not been merged yet.

Not merged code is prone to change and if one decides to start working over other's, on going, work one may be responsible to keep track of updates because the base branch will likely have changes 🤷‍♂️

Not blaming anyone, is the only thing that can be done now, just pointing out the real problem

Comment (gbudau) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: Squash and merge, or rebase and merge, are incompatible with stacked PRs. We have to become git wizards to solve merge conflicts.

Stacking PRs + merge commits are very beneficial and, in my opinion, could work better with the way we work asynchronously.
We can do small PRs, and if one of the previous PRs takes more time to review or merge, we can continue working on other PRs without too many merge conflicts.

I think we are gaining history readability at the cost of velocity.

Comment (Damrod) [reply transcendence-app/src/shared/relationship/relationship.module.ts:10]: fportela said
> Si te soy sincero tengo claro que es el squash merge o el merge commit. Lo lei en su día pero ya no me acuerdo. Pero en cualquier caso, si sirve para ahorrar problemas, a costa de trazabilidad, por mi no hay fallo 
I wish he had said so in public. Anyhow, I proceed to enable merge commits into master.

**Comment by gbudau** [comment]
Thanks!

**Pull request merged**
